source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,346 @@ Sorts Builders
+``Sorts.metaTextScore()`` static factory method to retrieve a sort specification
+for a descending sort by score (best match first).
+
+.. note::
+
+  If your MongoDB instance is running MongoDB 4.2 and earlier,
+  to use a ``Sorts.metaTextScore()``
+  expression with ``FindIterable.sort()``, you must also
+  include the same expression in a projection. This is demonstrated
+  in the code example below.
+
+The ``$text`` command performs a text search on the content indexed by the 
+:manual:`text index </core/index-text/>` of your collection. You must create
+a text index on your collection before you can run a ``$text`` query and use
+``Sorts.metaTextScore()``.
+
+.. example::
+
+   In the following code example, we show how to use the
+   ``metaTextScore()`` method to sort the results of a ``$text``
+   query on
+   :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`.
+   The following code snippet works as follows:
+
+   #. Create a text index for
+      :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`
+      on the ``food`` field.
+   #. Run a ``$text`` search for the phrase "maple donut".
+   #. Project ``metaTextScore()`` into your query results as the
+      ``score`` field. This step is optional if your MongoDB instance is
+      running MongoDB 4.4 or later. 
+   #. Sort using ``metaTextScore()`` as a sort specification.
+
+   .. code-block:: java
+   
+      import static com.mongodb.client.model.Sorts.metaTextScore;
+      import static com.mongodb.client.model.Sorts.ascending;
+   
+      // <MongoCollection setup code here>
+      
+      // create a text index on the food field
+      collection.createIndex(Indexes.text("food"));
+      // create text score sort specification
+      Bson metaTextScoreSort = metaTextScore("score");
+      String searchTerm = "maple donut";
+      // build $text search query
+      Bson searchQuery = new Document("$text", new Document("$search", searchTerm));
+      collection.find(searchQuery)
+              // project the text score into documents on the "score" field (optional in MongoDB 4.4 or later)
+              .projection(metaTextScoreSort)
+              // sort the documents on the "score" field
+              .sort(metaTextScoreSort)
+              .into(results);
+      for (Document result : results) {
+          System.out.println(result.toJson());
+      }
+   
+   The output of the above code snippet should look something like this: 
+   
+   .. code-block:: json
+
+      {"_id": 6, "letter": "c", "food": "maple donut", "score": 1.5}
+      {"_id": 2, "letter": "a", "food": "donuts and coffee", "score": 0.75}
+      {"_id": 3, "letter": "a", "food": "maple syrup", "score": 0.75}
+
+For additional information on the ``Sorts`` class, see the 
+:java-docs:`Sorts class API documentation </apidocs/mongodb-driver-core/com/mongodb/client/model/Sorts.html>`.
+For additional information on the ``$text`` and ``$meta`` aggregation
```

Comment:
Question: aren't these operators available for use in both query operations as well as aggregation pipelines? Or is this sentence mentioning that the $text is a query operator and $meta is an aggregation one? Could you clear this up for the reader?

Reply:
Thanks for drawing my attention to this incorrect use of terminology.

After some research I think I have a better understanding of terms. I will clearly state that `$text` is a [query operator](https://docs.mongodb.com/manual/reference/operator/query/text/) and that  `$meta` is an [aggregation operator](https://docs.mongodb.com/manual/reference/operator/aggregation/meta/). While `$meta` and `$text` can both be used in aggregation pipelines and `find`, I think this might be going into too much detail for this paragraph. I will just clean up the terminology.  Fix: https://github.com/mongodb/docs-java/pull/59/commits/59a640f52384cc3f78a6d95732bc32b8e24f75eb

Going forward I will try to refine my understanding of all of these terms and use them correctly.